sklearn/datasets/covtype.py

```diff
@@ -15,29 +15,30 @@
 # License: BSD 3 clause
 
 from gzip import GzipFile
-from io import BytesIO
 import logging
 from os.path import exists, join
-try:
-    from urllib2 import urlopen
-except ImportError:
-    from urllib.request import urlopen
+from os import remove
 
 import numpy as np
 
 from .base import get_data_home
+from .base import _fetch_remote
+from .base import RemoteFileMetadata
 from ..utils import Bunch
 from .base import _pkl_filepath
 from ..utils.fixes import makedirs
```

Comment:
AFAICT, it looks like the only difference is that you have the `exist_ok` functionality for Python 2.7. Basically if you are not using `exist_ok=True` you can just use `os.makedirs`.

Having said that you could simplify:

```py
import os

if not os.path.exists(lfw_home):
    os.makedirs(lfw_home)
```

by:
```py
from sklearn.utils.fixes import  makedirs

makedirs(lfw_home, exist_ok)
```

I think we should try to stay focussed in this PR in order to release 0.19 sooner rather than later. I would advise to do this kind of cosmetic changes in a separate PR.